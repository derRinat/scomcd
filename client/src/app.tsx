import { useCallback, useState } from "react";
import { Form } from "./components/form";
import { PaymentPlan } from "./components/payment-plan";
import { calculate } from "./utils/api";
import { Box, CssBaseline, Typography, Alert } from "@mui/material";
import { useError } from "./hooks/use-error";

export const App = () => {
  const [data, setData] = useState<CalcData>();
  const { error, setError } = useError();

  const onSubmitHandler = useCallback(
    async (data: CalcInput) => {
      setData(undefined);
      try {
        const result = await calculate(data);
        setData(result);
      } catch {
        setError(
          "Es ist ein Fehler aufgetreten. Bitte versuche es sp√§ter nochmal"
        );
      }
    },
    [setError]
  );

  return (
    <>
      <CssBaseline />
      <Box
        justifyContent="center"
        alignItems="center"
        m={3}
        sx={{ maxWidth: 1500 }}
      >
        <Box display="flex">
          <Typography variant="h4" fontWeight={700} mb={3}>
            Tilgungsrechner
          </Typography>
        </Box>

        <Form onSubmit={onSubmitHandler} />
        {!!data && <PaymentPlan data={data} />}
        {!!error && (
          <Box mt={3}>
            <Alert severity="error">{error}</Alert>
          </Box>
        )}
      </Box>
    </>
  );
};
