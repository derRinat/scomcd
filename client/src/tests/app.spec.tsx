import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { App } from "../app";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("should render the app", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
