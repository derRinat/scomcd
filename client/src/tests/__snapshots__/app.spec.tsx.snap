// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-o2qga3"
  >
    <div
      class="MuiBox-root css-k008qs"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-4a3zqv-MuiTypography-root"
      >
        Tilgungsrechner
      </h4>
    </div>
    <form
      class="MuiBox-root css-1cl7zdz"
    >
      <div
        class="MuiStack-root css-1p5q5e5-MuiStack-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          max="1000000"
          min="1000"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Darlehensbetrag (in €)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              name="amount"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Darlehensbetrag (in €)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          max="10"
          min="1"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            Sollzins (%)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r1:"
              name="interestRate"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Sollzins (%)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          max="10"
          min="1"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            Tilgung (%)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              name="repaymentRate"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Tilgung (%)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          max="30"
          min="1"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r3:"
            id=":r3:-label"
          >
            Laufzeit (Jahre)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r3:"
              name="period"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Laufzeit (Jahre)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-e64tsz"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Berechnen
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
